homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,507 @@
+            if DPCODE_TEMP_SET not in self.tuya_device.function:
+                return DEFAULT_MIN_TEMP
+
+            function_temp_item = self.tuya_device.function.get(DPCODE_TEMP_SET)
+            if function_temp_item is None:
+                return DEFAULT_MIN_TEMP
+            temp_value = json.loads(function_temp_item.values)
+            temp_min = temp_value.get("min")
+            if temp_min is None:
+                return DEFAULT_MIN_TEMP
+            return temp_min * 1.0 / (10 ** temp_set_scal)
+
+        if DPCODE_TEMP_SET_F not in self.tuya_device.function:
+            return DEFAULT_MIN_TEMP
+
+        temp_value_temp_f = self.tuya_device.function.get(DPCODE_TEMP_SET_F)
+        if temp_value_temp_f is None:
+            return DEFAULT_MIN_TEMP
+        temp_value_f = json.loads(temp_value_temp_f.values)
+
+        temp_min_f = temp_value_f.get("min")
+        if temp_min_f is None:
+            return DEFAULT_MIN_TEMP
+
+        return temp_min_f * 1.0 / (10 ** temp_set_scal)
+
+    @property
+    def target_temperature_step(self) -> float | None:
+        """Return target temperature setp."""
+        if (
+            DPCODE_TEMP_SET not in self.tuya_device.status_range
+            and DPCODE_TEMP_SET_F not in self.tuya_device.status_range
+        ):
+            return 1.0
+        temp_set_value_range = json.loads(
+            self.tuya_device.status_range.get(
+                DPCODE_TEMP_SET if self.is_celsius() else DPCODE_TEMP_SET_F
+            ).values
+        )
+        step = temp_set_value_range.get("step")
+        if step is None:
+            return None
+
+        temp_set_scale = self.get_temp_set_scale()
+        if temp_set_scale is None:
+            return None
+
+        return step * 1.0 / (10 ** temp_set_scale)
+
+    @property
+    def target_humidity(self) -> int:
+        """Return target humidity."""
+        return int(self.tuya_device.status.get(DPCODE_HUMIDITY_SET, 0))
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac mode."""
+        if not self.tuya_device.status.get(DPCODE_SWITCH, False):
+            return HVAC_MODE_OFF
+        if DPCODE_MODE not in self.tuya_device.status:
+            return HVAC_MODE_OFF
+        if self.tuya_device.status.get(DPCODE_MODE) is not None:
+            return TUYA_HVAC_TO_HA[self.tuya_device.status[DPCODE_MODE]]
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_modes(self) -> list[str]:
+        """Return hvac modes for select."""
+        if DPCODE_MODE not in self.tuya_device.function:
+            return []
+        modes = json.loads(self.tuya_device.function.get(DPCODE_MODE, {}).values).get(
+            "range"
+        )
+
+        hvac_modes = [HVAC_MODE_OFF]
+        for tuya_mode, ha_mode in TUYA_HVAC_TO_HA.items():
+            if tuya_mode in modes:
+                hvac_modes.append(ha_mode)
+
+        return hvac_modes
+
+    @property
+    def preset_modes(self) -> list[str]:
```

Comment:
We haven't implemented `set_preset` service support. Should we not do that? The `preset_modes` property won't have any effect if we don't support `SUPPORT_PRESET_MODE`.

Reply:
fixed